torchao/experimental/kernels/mps/metal/int4mm.metal

```diff
@@ -0,0 +1,80 @@
+#include <metal_stdlib>
+using namespace metal;
+
+// dispatchThreads:MTLSizeMake(N, M, 1)
+
+kernel void weight_to_int4pack(constant int *W [[buffer(0)]],
+                               device uchar *outputData [[buffer(1)]],
+                               constant uint2 &sizes [[buffer(2)]],
+                               uint2 thread_index [[thread_position_in_grid]]) {
+  const uint N = sizes.x;
+  const uint K = sizes.y;
+  const uint n = thread_index.x; // 0..N-1
+  const uint k2 = thread_index.y; // 0..K/2-1
+  int32_t src_val0 = W[n * K + 2 * k2];
+  int32_t src_val1 = W[n * K + 2 * k2 + 1];
+  outputData[n * (K / 2) + k2] = (uint8_t(src_val1) << 4) | uint8_t(src_val0);
+}
+
+template<typename T, unsigned groupSize>
+kernel void int4pack_mm(
```

Comment:
No, we don't need any of the int1, int2 or int4. I left them there, because I wrote them first, and then I wrote divbit. And like I said earlier, my intent was to not delete any shaders from the metal folder, even after they get superseded by an improved version. But will delete to make things cleaner